src/mon/MDSMonitor.cc

```diff
@@ -248,16 +248,17 @@ bool MDSMonitor::preprocess_beacon(MMDSBeacon *m)
 
   // check privileges, ignore if fails
   MonSession *session = m->get_session();
-  if (!session)
-    goto out;
+  assert(session);
   if (!session->is_capable("mds", MON_CAP_X)) {
     dout(0) << "preprocess_beacon got MMDSBeacon from entity with insufficient privileges "
 	    << session->caps << dendl;
+    mon->no_reply(m);
```

Comment:
Are there other places that "goto out" and don't invoke no_reply()? Can we just add a label/cleanup thing that they all use?